Fix issue where hex package does not build

lawik opened the pull request:
The directory `ports` is not one of the default directories that
hex will include:
https://hexdocs.pm/hex/Mix.Tasks.Hex.Build.html#module-package-configuration

This commit changes that to use src which is a default.

lawik commented:
Another option would be to extend the list of files. But I've generally seen Nerves projects use the src dir. Maybe that's primarily for NIFs but I'm not sure.

lawik commented:
Resolves #2

sasa1977 commented:
Thanks for the report and the fix! As soon as I saw it I suspected that the problem is the `ports` folder. My feeling is that `src` should be used for Erlang source files, so I think it would be better to explicitly include the ports folder in the list. Something like:

```elixir
defp package() do
  [
    ...,
    files: ~w(lib priv .formatter.exs mix.exs README* readme* LICENSE* license* CHANGELOG* 
              changelog* src c_src Makefile* ports)
  ]
end
```

should do the trick. Do you want to do this, or will you leave it to me?

lawik commented:
I can do that.

lawik committed:
Fix issue where hex package does not build

The directory ports is not one of the default directories that
hex will include:
https://hexdocs.pm/hex/Mix.Tasks.Hex.Build.html#module-package-configuration

Changed to explicitly listing the files.

lawik commented:
Was going to not remove the unused defaults, the c_src, etc but `mix hex.build` didn't like when those didn't resolve to actual files.

lawik commented:
And yeah, `src` for Erlang source sounds right. It is `c_src` I've seen a bunch of in Nerves-land.

sasa1977 commented:
> Was going to not remove the unused defaults, the c_src, etc but `mix hex.build` didn't like when those didn't resolve to actual files.

Ugh, this is a bit annoying, but it's a hex things, so we can't do much about it. I mean, we could do `mkdir_p`, but let's leave it as it is.

lawik commented:
Planning on adding all the makefiles, C sources and erlangs in the future then? :) Nah, I get it.

sasa1977 commented:
I'll see if we can add a more reliable test. I think that  running `mix hex.build`, expanding the generated tar, and then using that as a path dep in `init_test.exs` might do the trick.

sasa1977 commented:
Thanks!

sasa1977 commented:
0.1.1 is pushed to hex, thanks again!

lawik commented:
My pleasure ✨
